File: backend/src/main/java/pl/lahcimdev/comaco/customer/domain/Customer.java
package pl.lahcimdev.comaco.customer.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;
import lombok.NoArgsConstructor;
import pl.lahcimdev.comaco.user.domain.User;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.validation.constraints.Size;
import java.util.List;

@Data
@NoArgsConstructor

@Entity
public class Customer extends User {

    private String companyName;

    private String companyStreet;

    private String companyPostalCode;

    private String companyCity;

    private String companyNip;

    private String companyRegon;

    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL)
    @JsonManagedReference
    @Size(min = 1)
    private List<CompanyContact> contacts;

    private String photo;

}

File: frontend/src/api/models/customer.ts
/* tslint:disable */
import { CompanyContact } from './company-contact';
import { Role } from './role';
export interface Customer {
  companyCity?: string;
  companyName?: string;
  companyNip?: string;
  companyPostalCode?: string;
  companyRegon?: string;
  companyStreet?: string;
  contacts?: Array<CompanyContact>;
  createdBy?: string;
  createdDate?: string;
  id?: number;
  password?: string;
  photo?: string;
  roles?: Array<Role>;
  userType?: 'EMPLOYEE' | 'CUSTOMER';
  username?: string;
}
